components/Wallet/Send.tsx

```diff
@@ -44,172 +35,85 @@ export const Send = () => {
 
   // Transaction states
   const [txState, setTxState] = useState<TxState>(TxState.FillingForm)
-  const [txError, setTxError] = useState<Error | null>(null)
+  const [txFee, setTxFee] = useState<FilecoinNumber | null>(null)
 
-  // Placeholder message for getting gas params
-  const [message, setMessage] = useState<Message | null>(null)
-
-  // Prevent redundant updates to message so that we don't
-  // invoke the useGetGasParams hook more than necessary
-  const setMessageIfChanged = () => {
-    if (!isToAddressValid || !isValueValid || !isParamsValid) {
-      setMessage(null)
-      return
-    }
-    if (
-      !message ||
-      message.to !== toAddress ||
-      message.value.toString() !== value.toAttoFil() ||
-      message.params !== params
-    )
-      setMessage(
-        new Message({
-          to: toAddress,
-          from: wallet.address,
-          nonce: 0,
-          value: value.toAttoFil(),
-          method: 0,
-          params: params,
-          gasPremium: 0,
-          gasFeeCap: 0,
-          gasLimit: 0
-        })
-      )
-  }
-
-  // Max transaction fee used for getting gas params. Will be
-  // null until the user manually changes the transaction fee.
-  const [maxFee, setMaxFee] = useState<FilecoinNumber | null>(null)
-
-  // Load gas parameters when message or max fee changes
-  const {
-    gasParams,
-    loading: gasParamsLoading,
-    error: gasParamsError
-  } = useGetGasParams(walletProvider, message, maxFee)
+  // Create message from input
+  const message = useMemo<Message | null>(
+    () =>
+      isToAddressValid && isValueValid && isParamsValid && value
```

Comment:
Should we be using some `isValueValid` here instead of `value`?

Reply:
Here it's checking both `isValueValid` and `value`.

`isValueValid` needs to be checked so we don't create a `Message` when value is out of range (min/max).
`value` needs to be checked because `setIsValid` is called separately from `onChange`. This creates a render state where `isValueValid` is true, but `value` is still `null`. That causes an error when calling `value.toAttofil`